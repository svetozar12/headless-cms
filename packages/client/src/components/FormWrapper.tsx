import React from "react";

interface IFormWrapper {
  children: React.ReactNode;
  error: string;
}

const FormWrapper = (props: IFormWrapper) => {
  const { children, error } = props;
  return (
    <div className="flex h-screen w-full items-center justify-center bg-mainBlack ">
      <form className="flex h-3/5 w-5/6 flex-col justify-center gap-3 rounded-md bg-offBlack py-5 px-16 md:w-2/4 xl:w-1/4">
        {error && <h1 className="text-center text-xl text-red-500">{error}</h1>}
        {children}
      </form>
    </div>
  );
};

export default FormWrapper;
