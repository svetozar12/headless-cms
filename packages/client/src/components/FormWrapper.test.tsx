import type { ReactWrapper } from "enzyme";
import { mount } from "enzyme";
import { beforeEach, describe, expect, it, vi } from "vitest";
import FormWrapper from "./FormWrapper";

describe("Button component", () => {
  let wrapper: ReactWrapper;
  const mockedOnSubmit = vi.fn();

  beforeEach(() => {
    mockedOnSubmit.mockReset();

    wrapper = mount(
      <FormWrapper error={""}>
        <p>hello</p>
      </FormWrapper>
    );
  });
  it("should render without error", () => {
    expect(wrapper.find("form").length).toBe(1);
    expect(wrapper.find("p").length).toBe(1);
    expect(wrapper.find("p").text()).toBe("hello");
  });
  it("should render with error", () => {
    wrapper.setProps({ error: "error" });
    expect(wrapper.find("form").length).toBe(1);
    expect(wrapper.find("p").length).toBe(1);
    expect(wrapper.find("p").text()).toBe("hello");
    expect(wrapper.find("h1").text()).toBe("error");
    expect(wrapper).toMatchSnapshot();
  });
});
