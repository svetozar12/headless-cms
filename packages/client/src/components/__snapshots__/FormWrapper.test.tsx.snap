// Vitest Snapshot v1

exports[`Button component > should render with error 1`] = `ReactWrapper {}`;

exports[`Button component > should render without error 1`] = `ReactWrapper {}`;
